src/core/DoomInPlayCounter.ttslua

```diff
@@ -1,132 +1,105 @@
+local originAndSize = {
+    { origin = { -55, ypos, 16.5 }, size = { 12, 1, 25 } },
+    { origin = { -55, ypos, -16.5 }, size = { 12, 1, 25 } },
+    { origin = { -25, ypos, 27 }, size = { 25, 1, 12 } },
+    { origin = { -25, ypos, -27 }, size = { 25, 1, 12 } }
+}
+
+-- create button, context menu and start loop
+function onLoad()
     self.createButton({
-      label=tostring(val),
-      click_function="add_subtract",
-      function_owner=self,
-      position={0,0.05,0},
-      height=600,
-      width=1000,
-      alignment = 3,
-      scale={x=1.5, y=1.5, z=1.5},
-      font_size=600,
-      font_color=f_color,
-      color={0,0,0,0}
-      })
-
-
-
-
-    if light_mode then
-        lightButtonText = "[ Set dark ]"
-    else
-        lightButtonText = "[ Set light ]"
-    end
-
-end
-
-function removeAll()
-    self.removeInput(0)
-    self.removeInput(1)
-    self.removeButton(0)
-    self.removeButton(1)
-    self.removeButton(2)
-end
-
-function reloadAll()
-    removeAll()
-    createAll()
-
-    updateSave()
+        label = tostring(0),
+        click_function = "moreInformation",
+        function_owner = self,
+        position = { 0, 0.06, 0 },
+        height = 600,
+        width = 1000,
+        scale = { 1.5, 1.5, 1.5 },
+        font_size = 600,
+        font_color = { 1, 1, 1, 100 },
+        color = { 0, 0, 0, 0 }
+    })
+
+    -- context menu
+    self.addContextMenuItem("More Information", moreInformation)
+    self.addContextMenuItem("Remove Doom", function()
+        removeDoom = true
+        Wait.stop(loopID)
+        countDoom()
+        Wait.time(function()
+            removeDoom = false
+            loopID = Wait.time(countDoom, 2, -1)
+        end, 2)
+    end)
+
+    loopID = Wait.time(countDoom, 2, -1)
 end
 
-function swap_fcolor(_obj, _color, alt_click)
-    light_mode = not light_mode
-    reloadAll()
+function moreInformation()
+    printToAll("------------------------------", "White")
+    printToAll("Doom-in-Play Counter v" .. information["version"] .. " by Chr1Z", "Orange")
+    printToAll("last updated: " .. information["last_updated"], "White")
+    printToAll("Automatically counts the doom in play (exluding the agenda and objects with the ignore tag).", "Green")
+    printToAll("ignore tag: " .. IGNORE_TAG, "White")
 end
 
-function swap_align(_obj, _color, alt_click)
-    center_mode = not center_mode
-    reloadAll()
+-- main function
+function countDoom()
+    local doom = 0
+    for i = 1, 5 do doom = doom + search(i) end
+    self.editButton({ index = 0, label = tostring(doom) })
 end
 
-function editName(_obj, _string, value)
-    self.setName(value)
-    setTooltips()
-end
+-- searches playermats (num = 1-4) or the scripting zone (num = 5)
+function search(num)
+    local val = 0
+    if num == 5 then
+        for _, obj in ipairs(zone.getObjects()) do
+            val = val + getResult(obj)
+        end
+    else
+        CAST_PARAMETERS.origin = originAndSize[num].origin
```

Comment:
UPPER_SNAKE implies a constant, even if Lua doesn't enforce it changing the values is bad form (and can carry over).  Make a new local table that's initialized to the values from CAST_PARAMETERS, and change that as needed.

Reply:
changed capitalization